libcoz/libcoz.cpp

```diff
@@ -359,6 +381,32 @@ extern "C" {
     return real::pthread_rwlock_unlock(rwlock);
   }
 
+  void * malloc(size_t bytes_requested) throw() {
+    if (!initialized && !init_in_progress) {
+      static LocalArena<1 << 22> arena;
```

Comment:
For most programs that is too large memeory heap that could affect its performance in weird ways. Yet for some exotic programs, that allocate large amount of memory before main, that might be too little. It would be better to use your own allocator only during `dlsym` calls that create problems.

Reply:
Yep, `dlsym` should be enough. I'll check your branch on my application sometime in future